### PR TITLE
Fix _record_gradient attribute error in tensorflow>=2.5.0

### PR DESCRIPTION
**Pull request by filusn**
The fix ensures compatibility of module with tensorflow>=2.5.0, where the **_record_gradient** was replaced with **record_gradient** in _tensorflow.python.eager.backprop_.

### CODE REVIEW

**Comment by filusn** [comment]
Issue #2189

**Review by laneciar — APPROVED** [review]
Fixed my issue, now stuck with a ton of gradient registry has no entry for some reason.

**Comment by abhirajD** [comment]
> Fixed my issue, now stuck with a ton of gradient registry has no entry for some reason.

Same.

The build seems to be failing as well (appveyor). How does one go about fixing it? Or its better to be left alone for now?

**Comment by laneciar** [comment]
> > Fixed my issue, now stuck with a ton of gradient registry has no entry for some reason.
> 
> Same.
> 
> The build seems to be failing as well (appveyor). How does one go about fixing it? Or its better to be left alone for now?

The following is kind of tedious but fixed it for me:
`    shap.explainers._deep.deep_tf.op_handlers["AddV2"] = shap.explainers._deep.deep_tf.passthrough
    shap.explainers._deep.deep_tf.op_handlers["BatchMatMulV2"] = shap.explainers._deep.deep_tf.passthrough
    shap.explainers._deep.deep_tf.op_handlers["TensorListStack"] = shap.explainers._deep.deep_tf.passthrough
    shap.explainers._deep.deep_tf.op_handlers["StatelessWhile"] = shap.explainers._deep.deep_tf.passthrough
    shap.explainers._deep.deep_tf.op_handlers["TensorListFromTensor"] = shap.explainers._deep.deep_tf.passthrough
    shap.explainers._deep.deep_tf.op_handlers["Sin"] = shap.explainers._deep.deep_tf.passthrough
`
So pretty much adding any handlers that error out

**Review by nickkxsper — APPROVED** [review]
This worked for me

**Comment by INF800** [comment]
> > > Fixed my issue, now stuck with a ton of gradient registry has no entry for some reason.
> > 
> > 
> > Same.
> > The build seems to be failing as well (appveyor). How does one go about fixing it? Or its better to be left alone for now?
> 
> The following is kind of tedious but fixed it for me: ` shap.explainers._deep.deep_tf.op_handlers["AddV2"] = shap.explainers._deep.deep_tf.passthrough shap.explainers._deep.deep_tf.op_handlers["BatchMatMulV2"] = shap.explainers._deep.deep_tf.passthrough shap.explainers._deep.deep_tf.op_handlers["TensorListStack"] = shap.explainers._deep.deep_tf.passthrough shap.explainers._deep.deep_tf.op_handlers["StatelessWhile"] = shap.explainers._deep.deep_tf.passthrough shap.explainers._deep.deep_tf.op_handlers["TensorListFromTensor"] = shap.explainers._deep.deep_tf.passthrough shap.explainers._deep.deep_tf.op_handlers["Sin"] = shap.explainers._deep.deep_tf.passthrough` So pretty much adding any handlers that error out

why not use 
```
for k in shap.explainers._deep.deep_tf.op_handlers.keys():
    shap.explainers._deep.deep_tf.op_handlers[k] = shap.explainers._deep.deep_tf.passthrough 
```

**Comment by rozeappletree** [comment]
> > > Fixed my issue, now stuck with a ton of gradient registry has no entry for some reason.
> > 
> > 
> > Same.
> > The build seems to be failing as well (appveyor). How does one go about fixing it? Or its better to be left alone for now?
> 
> The following is kind of tedious but fixed it for me: ` shap.explainers._deep.deep_tf.op_handlers["AddV2"] = shap.explainers._deep.deep_tf.passthrough shap.explainers._deep.deep_tf.op_handlers["BatchMatMulV2"] = shap.explainers._deep.deep_tf.passthrough shap.explainers._deep.deep_tf.op_handlers["TensorListStack"] = shap.explainers._deep.deep_tf.passthrough shap.explainers._deep.deep_tf.op_handlers["StatelessWhile"] = shap.explainers._deep.deep_tf.passthrough shap.explainers._deep.deep_tf.op_handlers["TensorListFromTensor"] = shap.explainers._deep.deep_tf.passthrough shap.explainers._deep.deep_tf.op_handlers["Sin"] = shap.explainers._deep.deep_tf.passthrough` So pretty much adding any handlers that error out

These are not working from me in tf 2.6.2

**Comment by rozeappletree** [comment]
I tried using 
`
!pip install git+git://github.com/filusn/shap.git@e6f842573943332d6bcc941f46d3c96dc2ac793f
` 

but got the same error for `shap_values = e.shap_values(correct[1:5])`

**Comment by Yo-B** [comment]
worked for me too

**Comment by rozeappletree** [comment]
> I tried using `!pip install git+git://github.com/filusn/shap.git@e6f842573943332d6bcc941f46d3c96dc2ac793f`
> 
> but got the same error for `shap_values = e.shap_values(correct[1:5])`

There was an issue related to kaggle kernel. Worked in colab for me. Apologies for inconvenience!

**Comment by slundberg** [comment]
Thanks for the fix! Sorry I was slow on the review (was out for a while after paternity leave).

**Comment by filusn** [comment]
> Thanks for the fix! Sorry I was slow on the review (was out for a while after paternity leave).

Thank you for merging and all the best for you and your family!

**Comment by Guido-Hwang** [comment]
> This worked for me

me too
